packages/edit-site/src/components/dataviews/add-filter.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	privateApis as componentsPrivateApis,
+	Button,
+	BaseControl,
+	Icon,
+} from '@wordpress/components';
+import { chevronRightSmall } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const {
+	DropdownMenuV2,
+	DropdownSubMenuV2,
+	DropdownSubMenuTriggerV2,
+	DropdownMenuItemV2,
+} = unlock( componentsPrivateApis );
+
+export default function AddFilter( { filters, onChangeView } ) {
+	return (
+		<BaseControl>
```

Comment:
Why do we need this `BaseControl` wrapper?

Reply:
This is how it looks like without it (alignment issues):

<img width="589" alt="Captura de ecrã 2023-11-10, às 15 28 53" src="https://github.com/WordPress/gutenberg/assets/583546/a881c46d-a42a-435d-945a-be725df05e85">

Happy to solve it any other way.

Reply:
BaseControl is mean to add a label and a help to controls (inputs, selects...), if it's the spacing that is the issue here, we should probably just use a `HStack` with align center or something for the row or something like that.

Reply:
I've already tried setting the alignment in the [existing](packages/edit-site/src/components/dataviews/add-filter.js) HStack component, though it didn't work.

Reply:
Note the alignment of the components in the top-level is an issue for the "view actions" component as well:

<img width="487" alt="Captura de ecrã 2023-11-14, às 11 28 18" src="https://github.com/WordPress/gutenberg/assets/583546/359f59b8-5fc3-4410-84a4-24d5f2d03e85">

Reply:
The issue here is a mix of things: different height for components (search was 40, but others were 32) + different spacing (search, as a base control, uses a 8px margin-bottom, but the buttons use 6 padding top&bottom).

I've got a fix here https://github.com/WordPress/gutenberg/pull/55992/commits/1e99ee9f0ebad74269c62ace233377521c3cf3d0 Would this alternative be preferred? I'd still need to tweak the filters (`SelectControl`) and the view menu as well.

Reply:
Found a way to reset the `BaseControl` margin via component prop at https://github.com/WordPress/gutenberg/pull/55992/commits/353ca409f4f5c6d16d9877570bce07c77162b108 I'll mark this as fixed.